From 650a2ee051a89bc32f8c671081efa2ab875bc89d Mon Sep 17 00:00:00 2001
From: Reshetnikov Alexandr <hemn.still@gmail.com>
Date: Fri, 11 Aug 2023 19:54:38 +0400
Subject: [PATCH] static

---
 CPP/7zip/7zip_gcc.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index 9588f05..9327757 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -51,6 +51,8 @@ endif
 endif
 endif
 
+LDFLAGS_STATIC = -static
+
 LDFLAGS_STATIC = -DNDEBUG $(LDFLAGS_STATIC_2)
 
 ifndef O
-- 
2.37.2.windows.2

