diff --git "a/CPP/7zip/7zip_gcc.mak" "b/CPP/7zip/7zip_gcc.mak"
index 42fff6c..5af2c57 100644
--- "a/CPP/7zip/7zip_gcc.mak"
+++ "b/CPP/7zip/7zip_gcc.mak"
@@ -51,6 +51,8 @@ endif
 endif
 endif
 
+LDFLAGS_STATIC_2 = -static
+
 LDFLAGS_STATIC = -DNDEBUG $(LDFLAGS_STATIC_2)
 
 ifndef O
@@ -94,7 +96,7 @@ endif
 
 PROGPATH = $(O)/$(PROG)$(SHARED_EXT)
 PROGPATH_STATIC = $(O)/$(PROG)s$(SHARED_EXT)
-	
+
 ifdef IS_MINGW
 
 RM = del
@@ -106,7 +108,7 @@ CXXFLAGS_EXTRA = -DUNICODE -D_UNICODE
 # -Wno-delete-non-virtual-dtor
 
 DEL_OBJ_EXE = -$(RM) $(O)\*.o $(O)\$(PROG).exe $(O)\$(PROG).dll
- 
+
 else
 
 RM = rm -rf
@@ -1289,7 +1291,7 @@ $O/sha512.o: ../../../../C/hashes/sha512.c
 $O/blake3.o: ../../../../C/hashes/blake3.c
 	$(CC) $(CFLAGS) $<
 
-# Build MT API 
+# Build MT API
 $O/lz4-mt_common.o: ../../../../C/zstdmt/lz4-mt_common.c
 	$(CC) $(CFLAGS) $<
 $O/lz4-mt_compress.o: ../../../../C/zstdmt/lz4-mt_compress.c
@@ -1321,10 +1323,11 @@ $O/libzstd.$(DYSUFFIX): ../../../../C/zstd/lib/zstd.h
 	$(RM) zstd_build
 	$(MY_MKDIR) -p zstd_build
 	$(CD) zstd_build; \
-	cmake ../../../../../C/zstd/build/cmake -DCMAKE_BUILD_TYPE=Release -DZSTD_BUILD_PROGRAMS=OFF -DZSTD_BUILD_STATIC=OFF -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
+	cmake ../../../../../C/zstd/build/cmake -DCMAKE_BUILD_TYPE=Release -DZSTD_BUILD_PROGRAMS=OFF -DZSTD_BUILD_STATIC=ON -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
 	$(MAKE) -j; \
 	$(CD) ../; \
 	$(CP) zstd_build/lib/libzstd.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) zstd_build/lib/libzstd.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 # Compile zstd method and Handler
 $O/ZstdDecoder.o: ../../Compress/ZstdDecoder.cpp
@@ -1345,6 +1348,7 @@ $O/liblz4.$(DYSUFFIX): ../../../../C/lz4/lib/lz4.h
 	$(MAKE) -j; \
 	$(CD) ../; \
 	$(CP) lz4_build/liblz4.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) lz4_build/liblz4.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 # Compile lz4 method and Handler
 $O/Lz4Decoder.o: ../../Compress/Lz4Decoder.cpp
@@ -1367,10 +1371,13 @@ $O/libbrotlicommon.$(DYSUFFIX) $O/libbrotlienc.$(DYSUFFIX) $O/libbrotlidec.$(DYS
 	$(CP) brotli_build/libbrotlicommon.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 	$(CP) brotli_build/libbrotlidec.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 	$(CP) brotli_build/libbrotlienc.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) brotli_build/libbrotlicommon-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) brotli_build/libbrotlidec-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) brotli_build/libbrotlienc-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 	patchelf --force-rpath --set-rpath '$$ORIGIN/' $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlidec.$(DYSUFFIX)*
 	patchelf --force-rpath --set-rpath '$$ORIGIN/' $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlienc.$(DYSUFFIX)*
 
-# Compile brotli method and Handler 
+# Compile brotli method and Handler
 $O/BrotliDecoder.o: ../../Compress/BrotliDecoder.cpp
 	$(CXX) $(CXXFLAGS) $< -I ../../../../C/brotli/c/include
 $O/BrotliEncoder.o: ../../Compress/BrotliEncoder.cpp
