diff --git "a/CPP/7zip/7zip_gcc.mak" "b/CPP/7zip/7zip_gcc.mak"
index 42fff6c..f355db9 100644
--- "a/CPP/7zip/7zip_gcc.mak"
+++ "b/CPP/7zip/7zip_gcc.mak"
@@ -51,6 +51,8 @@ endif
 endif
 endif
 
+LDFLAGS_STATIC_2 = -static
+
 LDFLAGS_STATIC = -DNDEBUG $(LDFLAGS_STATIC_2)
 
 ifndef O
@@ -94,7 +96,7 @@ endif
 
 PROGPATH = $(O)/$(PROG)$(SHARED_EXT)
 PROGPATH_STATIC = $(O)/$(PROG)s$(SHARED_EXT)
-	
+
 ifdef IS_MINGW
 
 RM = del
@@ -106,7 +108,7 @@ CXXFLAGS_EXTRA = -DUNICODE -D_UNICODE
 # -Wno-delete-non-virtual-dtor
 
 DEL_OBJ_EXE = -$(RM) $(O)\*.o $(O)\$(PROG).exe $(O)\$(PROG).dll
- 
+
 else
 
 RM = rm -rf
@@ -192,15 +194,15 @@ endif
 7z_BIN = bin
 7z_LIB = lib
 ifeq ($(PROG), 7za)
-7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -llizard -llz5 -lfast-lzma2 -llzhamcomp -llzhamdecomp -llzhamdll
+7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -lfast-lzma2
 7z_ADDON_LIB_FLAG = -L./$(O)/$(7z_LIB)/$(7Z_ADDON_CODEC) -Wl,-rpath='$$ORIGIN'/../$(7z_LIB)/$(7Z_ADDON_CODEC)
 PROGPATH = $(O)/$(7z_BIN)/$(PROG)$(SHARED_EXT)
 else ifeq ($(PROG), 7zz)
-7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -llizard -llz5 -lfast-lzma2 -llzhamcomp -llzhamdecomp -llzhamdll
+7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -lfast-lzma2
 7z_ADDON_LIB_FLAG = -L./$(O)/$(7z_LIB)/$(7Z_ADDON_CODEC) -Wl,-rpath='$$ORIGIN'/../$(7z_LIB)/$(7Z_ADDON_CODEC)
 PROGPATH = $(O)/$(7z_BIN)/$(PROG)$(SHARED_EXT)
 else ifeq ($(PROG), 7z)
-7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -llizard -llz5 -lfast-lzma2 -llzhamcomp -llzhamdecomp -llzhamdll
+7z_ADDON_LIB = -lzstd -llz4 -lbrotlienc -lbrotlidec -lbrotlicommon -lfast-lzma2
 7z_ADDON_LIB_FLAG = -L./$(O)/$(7z_LIB)/$(7Z_ADDON_CODEC) -Wl,-rpath='$$ORIGIN'/$(7Z_ADDON_CODEC)
 PROGPATH = $(O)/$(7z_LIB)/$(PROG)$(SHARED_EXT)
 else
@@ -217,7 +219,7 @@ $(7Z_ADDON_CODEC):
 	$(MY_MKDIR) $(O)/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 LFLAGS_ALL = -s $(MY_ARCH_2) $(LDFLAGS) $(LD_arch) $(OBJS) $(MY_LIBS) $(LIB2)
-$(PROGPATH): $(ZSTD_LIB) $(LZ4_LIB) $(BROTLI_LIB) $(LIZARD_LIB) $(LZ5_LIB) $(FAST-LZMA2_LIB) $(LZHAM_LIB) $(OBJS)
+$(PROGPATH): $(ZSTD_LIB) $(LZ4_LIB) $(BROTLI_LIB) $(FAST-LZMA2_LIB) $(OBJS)
 	$(CXX) -o $(PROGPATH) $(LFLAGS_ALL) $(7z_ADDON_LIB) $(7z_ADDON_LIB_FLAG)
 
 $(PROGPATH_STATIC): $(OBJS)
@@ -1289,7 +1291,7 @@ $O/sha512.o: ../../../../C/hashes/sha512.c
 $O/blake3.o: ../../../../C/hashes/blake3.c
 	$(CC) $(CFLAGS) $<
 
-# Build MT API 
+# Build MT API
 $O/lz4-mt_common.o: ../../../../C/zstdmt/lz4-mt_common.c
 	$(CC) $(CFLAGS) $<
 $O/lz4-mt_compress.o: ../../../../C/zstdmt/lz4-mt_compress.c
@@ -1321,10 +1323,11 @@ $O/libzstd.$(DYSUFFIX): ../../../../C/zstd/lib/zstd.h
 	$(RM) zstd_build
 	$(MY_MKDIR) -p zstd_build
 	$(CD) zstd_build; \
-	cmake ../../../../../C/zstd/build/cmake -DCMAKE_BUILD_TYPE=Release -DZSTD_BUILD_PROGRAMS=OFF -DZSTD_BUILD_STATIC=OFF -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
+	cmake ../../../../../C/zstd/build/cmake -DCMAKE_BUILD_TYPE=Release -DZSTD_BUILD_PROGRAMS=OFF -DZSTD_BUILD_STATIC=ON -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
 	$(MAKE) -j; \
 	$(CD) ../; \
 	$(CP) zstd_build/lib/libzstd.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) zstd_build/lib/libzstd.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 # Compile zstd method and Handler
 $O/ZstdDecoder.o: ../../Compress/ZstdDecoder.cpp
@@ -1341,10 +1344,11 @@ $O/liblz4.$(DYSUFFIX): ../../../../C/lz4/lib/lz4.h
 	$(RM) lz4_build
 	$(MY_MKDIR) -p lz4_build
 	$(CD) lz4_build; \
-	cmake ../../../../../C/lz4/build/cmake -DCMAKE_BUILD_TYPE=Release -DLZ4_BUILD_CLI=OFF -DLZ4_BUILD_LEGACY_LZ4C=OFF -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
+	cmake ../../../../../C/lz4/build/cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_STATIC_LIBS=ON -DLZ4_BUILD_CLI=OFF -DLZ4_BUILD_LEGACY_LZ4C=OFF -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
 	$(MAKE) -j; \
 	$(CD) ../; \
 	$(CP) lz4_build/liblz4.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) lz4_build/liblz4.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 # Compile lz4 method and Handler
 $O/Lz4Decoder.o: ../../Compress/Lz4Decoder.cpp
@@ -1367,10 +1371,13 @@ $O/libbrotlicommon.$(DYSUFFIX) $O/libbrotlienc.$(DYSUFFIX) $O/libbrotlidec.$(DYS
 	$(CP) brotli_build/libbrotlicommon.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 	$(CP) brotli_build/libbrotlidec.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 	$(CP) brotli_build/libbrotlienc.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
+	$(CP) brotli_build/libbrotlicommon-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlicommon.a
+	$(CP) brotli_build/libbrotlidec-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlidec.a
+	$(CP) brotli_build/libbrotlienc-static.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlienc.a
 	patchelf --force-rpath --set-rpath '$$ORIGIN/' $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlidec.$(DYSUFFIX)*
 	patchelf --force-rpath --set-rpath '$$ORIGIN/' $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/libbrotlienc.$(DYSUFFIX)*
 
-# Compile brotli method and Handler 
+# Compile brotli method and Handler
 $O/BrotliDecoder.o: ../../Compress/BrotliDecoder.cpp
 	$(CXX) $(CXXFLAGS) $< -I ../../../../C/brotli/c/include
 $O/BrotliEncoder.o: ../../Compress/BrotliEncoder.cpp
@@ -1378,53 +1385,6 @@ $O/BrotliEncoder.o: ../../Compress/BrotliEncoder.cpp
 $O/BrotliRegister.o: ../../Compress/BrotliRegister.cpp
 	$(CXX) $(CXXFLAGS) $< -I ../../../../C/brotli/c/include
 
-# Build lizard lib static and dynamic
-$O/liblizard.$(DYSUFFIX): ../../../../C/lizard/lib/lizard_frame.h
-	$(MAKE) -C ../../../../C/lizard/lib
-	$(CP) ../../../../C/lizard/lib/liblizard.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
-
-# Compile lizard method and Handler
-$O/LizardDecoder.o: ../../Compress/LizardDecoder.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/LizardEncoder.o: ../../Compress/LizardEncoder.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/LizardRegister.o: ../../Compress/LizardRegister.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/LizardHandler.o: ../../Archive/LizardHandler.cpp
-	$(CXX) $(CXXFLAGS) $<
-
-# Build lz5 lib static and dynamic
-$O/liblz5.$(DYSUFFIX): ../../../../C/lz5/lib/lz5frame.h
-	$(MAKE) -C ../../../../C/lz5/lib
-	$(CP) ../../../../C/lz5/lib/liblz5.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
-
-# Compile lz5 method and Handler
-$O/Lz5Decoder.o: ../../Compress/Lz5Decoder.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/Lz5Encoder.o: ../../Compress/Lz5Encoder.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/Lz5Register.o: ../../Compress/Lz5Register.cpp
-	$(CXX) $(CXXFLAGS) $<
-$O/Lz5Handler.o: ../../Archive/Lz5Handler.cpp
-	$(CXX) $(CXXFLAGS) $<
-
-# Build lzham lib static and dynamic
-$O/liblzhamdll.$(DYSUFFIX) $O/liblzhamcomp.$(DYSUFFIX) $O/liblzhamdecomp.$(DYSUFFIX): ../../../../C/lzham_codec/lzhamdll/lzham_api.cpp
-	$(RM) lzham_build
-	$(MY_MKDIR) -p lzham_build
-	$(CD) lzham_build; \
-	cmake ../../../../../C/lzham_codec/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX); \
-	$(MAKE) -j; \
-	$(CD) ../; \
-	$(CP) lzham_build/lzhamcomp/liblzhamcomp.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
-	$(CP) lzham_build/lzhamdecomp/liblzhamdecomp.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
-	$(CP) lzham_build/lzhamdll/liblzhamdll.$(DYSUFFIX)* $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
-	patchelf --force-rpath --set-rpath '$$ORIGIN/' $O/$(7z_LIB)/$(7Z_ADDON_CODEC)/liblzhamdll.$(DYSUFFIX)*
-
-# Compile lzham register
-$O/LzhamRegister.o: ../../Compress/LzhamRegister.cpp
-	$(CXX) $(CXXFLAGS) $< -I ../../../../C/lzham_codec/include
-
 # Compile FastLzma2 method
 $O/libfast-lzma2.$(DYSUFFIX): ../../../../C/fast-lzma2/fast-lzma2.h
 	$(MAKE) -C ../../../../C/fast-lzma2
@@ -1435,7 +1395,7 @@ $O/libfast-lzma2.$(DYSUFFIX): ../../../../C/fast-lzma2/fast-lzma2.h
 # only for 7zr
 $O/libfast-lzma2.a: ../../../../C/fast-lzma2/fast-lzma2.h
 	$(MAKE) -C ../../../../C/fast-lzma2
-	$(CP) ../../../../C/fast-lzma2/libfast-lzma2.a $O
+	$(CP) ../../../../C/fast-lzma2/libfast-lzma2.a $O/$(7z_LIB)/$(7Z_ADDON_CODEC)
 
 $O/FastLzma2Register.o: ../../Compress/FastLzma2Register.cpp
 	$(CXX) $(CXXFLAGS) $<
@@ -1469,7 +1429,4 @@ clean:
 	$(RM) zstd_build
 	$(RM) lz4_build
 	$(RM) brotli_build
-	$(RM) lzham_build
 	$(RM) b
-	$(MAKE) -C ../../../../C/lizard/lib clean
-	$(MAKE) -C ../../../../C/lz5/lib clean
