From 83965d8dd1e6e802c46febc204cddd32fbecea8a Mon Sep 17 00:00:00 2001
From: Reshetnikov Alexandr <hemn.still@gmail.com>
Date: Wed, 15 May 2024 20:56:13 +0400
Subject: [PATCH] LDFLAGS_STATIC2 = -static

---
 CPP/7zip/7zip_gcc.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index b3f94be..d3b5066 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -85,6 +85,8 @@ endif
 endif
 endif
 
+LDFLAGS_STATIC_2 = -static
+
 LDFLAGS_STATIC = $(CFLAGS_DEBUG) $(LDFLAGS_STATIC_2) $(LDFLAGS_STATIC_3)
 
 ifndef O
-- 
2.43.0.windows.1

